tutorials/torchao_vllm_integration.md

```diff
@@ -0,0 +1,465 @@
+# TorchAO Integration with VLLM: Architecture and Usage Guide
+
+This tutorial provides a comprehensive overview of how TorchAO integrates with VLLM, and what needs to be implemented to have a new technique work E2E.
+
+## Table of Contents
+* [Configuration System](#configuration-system)
+* [Usage Examples](#usage-examples)
+* [Supported Quantization Methods](#supported-quantization-methods)
+* [Advanced Features](#advanced-features)
+* [Performance Considerations](#performance-considerations)
+* [Integration Architecture](#integration-architecture)
+
+
+## Configuration System
+
+### 1. HuggingFace Model Configuration
+
+TorchAO quantization is configured through the model's `config.json` file:
+
+```json
+{
+  "model_type": "llama",
+  "quant_type": {
+    "default": {
+      "_type": "Int4WeightOnlyConfig",
+      "_data": {
+        "group_size": 128,
+        "use_hqq": true
+      }
+    }
+  }
+}
+```
+
+### 2. TorchAO Configuration Classes
+
+All quantization methods inherit from `AOBaseConfig`:
+
+```python
+from torchao.core.config import AOBaseConfig
+from torchao.quantization import Int4WeightOnlyConfig
+
+# Example configuration
+config = Int4WeightOnlyConfig(
+    group_size=128,
+    use_hqq=True,
+)
+assert isinstance(config, AOBaseConfig)
+```
+
+### 3. Module-Level Configuration
+
+For granular control, use `ModuleFqnToConfig`:
+
+```python
+from torchao.quantization import ModuleFqnToConfig, Int4WeightOnlyConfig, Int8WeightOnlyConfig
+
+config = ModuleFqnToConfig({
+    "model.layers.0.self_attn.q_proj": Int4WeightOnlyConfig(group_size=64),
+    "model.layers.0.self_attn.k_proj": Int4WeightOnlyConfig(group_size=64),
+    "model.layers.0.mlp.gate_proj": Int8WeightOnlyConfig(),
+    "_default": Int4WeightOnlyConfig(group_size=128)  # Default for other modules
+})
+```
+
+## Usage Examples
+
+### 1. Quantizing Models with HuggingFace Integration
+
+```python
+from transformers import TorchAoConfig, AutoModelForCausalLM
+from torchao.quantization import Int4WeightOnlyConfig
+
+# Create quantization configuration
+quantization_config = TorchAoConfig(
+    quant_type=Int4WeightOnlyConfig(group_size=128, use_hqq=True)
```

Comment:
Do we have a list of configs that are currently supported through `TorchAoConfig`?